ts/modules/full-screen.src.ts

```diff
@@ -18,63 +15,169 @@ import H from '../parts/Globals.js';
  */
 declare global {
     namespace Highcharts {
+        interface Chart {
+            browserProps: {
+                fullscreenChange: string;
+                requestFullscreen: string;
+                exitFullscreen: string;
+            };
+            isFullscreen?: boolean;
+            /** @requires modules/full-screen */
+            closeFullscreen(): void;
+            /** @requires modules/full-screen */
+            openFullscreen(): void;
+            /** @requires modules/full-screen */
+            toggleFullscreen(): void;
+            unbindFullscreenEvent?: Function;
+        }
         class FullScreen {
             public constructor(container: HTMLDOMElement);
             public init(container: HTMLDOMElement): void;
         }
     }
 }
 
+/**
+ * The module allows user to enable display chart in full screen mode.
+ * Used in StockTools too.
+ * Based on default solutions in browsers.
+ *
+ */
+
+/**
+ * The flag is set to `true` if the chart is displayed in fullscreen mode.
+ *
+ * @name Highcharts.Chart#isFullscreen
+ * @type {boolean|undefined}
+ * @since next
+ */
+
+// Helping function - to open fullscreen, use toggleFullscreen() instead.
+H.Chart.prototype.openFullscreen = function (this: Highcharts.Chart): void {
+    const chart = this,
+        menuItems = chart.options.exporting?.buttons?.contextButton.menuItems;
+
+    // Handle closeFullscreen() method when user clicks 'Escape' button.
+    chart.unbindFullscreenEvent = H.addEvent(
+        chart.container.ownerDocument, // chart's document
+        chart.browserProps.fullscreenChange,
+        function (): void {
+            // Handle lack of async of browser's fullScreenChange event.
+            if (chart.isFullscreen) {
+                chart.isFullscreen = false;
+                chart.closeFullscreen();
+            } else {
+                chart.isFullscreen = true;
+            }
+        }
+    );
+
+    (chart as any).container.parentNode[chart.browserProps.requestFullscreen]();
```

Comment:
parentNode needs a check agains `instanceof Element`, so that you can use the full screen function type safe.
```
    if (chart.container.parentNode instanceof Element) {
        chart.container.parentNode[chart.browserProps.requestFullscreen]();
    }
```

Reply:
Thanks, I made changes, could you take a look once again? Especially at Globals.ts and `instanceof Document`